src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentTrainEnumerator.cs

```diff
@@ -153,6 +155,12 @@ public CoraxDocumentTrainEnumerator(TransactionOperationContext indexContext, Co
         _take = take;
         _token = token;
 
+        // RavenDB-21043: Tracking the total memory allocated by the thread is also a way to limit the total resources allocated
+        // to the training process. We are currently limiting the default to 2Gb and we haven't seen any deterioration in the 
+        // compression using that limit. However, given there is no limitation in 64bits mode, we could increase it if we find
+        // cases which are not covered.
+        _maxAllocatedMemory = maxAllocatedMemory == default ? new Size(2, SizeUnit.Gigabytes) : maxAllocatedMemory;
```

Comment:
> If we ever find on the field that this is better to be modifiable, it is a very simple change to enable that.

I'd prefer to introduce a configuration option for that right now. With different default values for x64 and x86. We already do it in couple of places, for example:

https://github.com/ravendb/ravendb/blob/0e91c6ce2a71dd21b16af4190ae29afb7d31e51c/src/Raven.Server/Config/Categories/IndexingConfiguration.cs#L35-L37